data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -46,265 +49,64 @@ object UserMetadataUpdater {
-    if (!framework.isEmpty) {
-      val board = framework.getOrDefault("board", List().asJava)
-      val medium = framework.getOrDefault("medium", List().asJava)
-      val grade = framework.getOrDefault("gradeLevel", List().asJava)
-      val subject = framework.getOrDefault("subject", List().asJava)
-      val id = framework.getOrDefault("id", List().asJava)
-      result.+=("board" -> board, "medium" -> medium, "grade" -> grade, "subject" -> subject, "framework" -> id)
-    }
-    result
-  }
-
-  /**
-   *
-   * @param rootOrgId
-   * @param custRootOrgId
-   * @return true/false
-   */
-  def isCustodianUser(rootOrgId: String, custRootOrgId: String): Boolean = {
-    if (rootOrgId == custRootOrgId) true else false
-  }
 
-  def getCustodianUserInfo(metrics: Metrics, userId: String, config: UserCacheUpdaterConfigV2, cassandraConnect: CassandraUtil): mutable.Map[String, AnyRef] = {
-    var userExternalInfo: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
-
-    val userDeclarationList = readUserDeclaredInfo(userId, config, cassandraConnect, metrics)
-    if (!userDeclarationList.isEmpty) {
-      val userDeclarationInfo = userDeclarationList.stream().filter(f => f.getString("persona").equalsIgnoreCase(config.personaType)).findFirst().orElse(null)
-      if (null != userDeclarationInfo) {
-        val orgId = userDeclarationInfo.getString("orgid")
-        val validOrgId = validateOrgId(orgId, config, cassandraConnect, metrics)
-
-        if (validOrgId) {
-          val columnDefinitions = userDeclarationInfo.getColumnDefinitions()
-          val columnCount = columnDefinitions.size
-          for (i <- 0 until columnCount) {
-            userExternalInfo.put(columnDefinitions.getName(i), userDeclarationInfo.getObject(i))
+      //Location and School Information
+      val locationInfo = response.userLocations
+      if(null != locationInfo && !locationInfo.isEmpty) {
+        locationInfo.forEach(location => {
+          location.getOrDefault("type", "").asInstanceOf[String].toLowerCase match {
+            case config.stateKey => userCacheData.put(config.stateKey, location.getOrDefault("name", "").asInstanceOf[String])
+            case config.districtKey => userCacheData.put(config.districtKey, location.getOrDefault("name", "").asInstanceOf[String])
+            case config.blockKey => userCacheData.put(config.blockKey, location.getOrDefault("name", "").asInstanceOf[String])
+            case config.clusterKey => userCacheData.put(config.clusterKey, location.getOrDefault("name", "").asInstanceOf[String])
+            case config.schoolKey => userCacheData.put(config.schoolNameKey, location.getOrDefault("name", "").asInstanceOf[String])
+              userCacheData.put(config.schoolUdiseCodeKey, location.getOrDefault("code", "").asInstanceOf[String])
+            case _ => //Do Nothing
```

Comment:
can this be generalized to put all keys? instead of whitelisting only few?

Reply:
Done.